Sw 99 - Catalogue WIP

AndreasBM1 opened the pull request:
Vi har lavet lidt på den heroppe

Todo:
Hent items fra db istedet for array på catalogue.component.ts
set up routing for id'er på vin (mayhaps)

MyMainManMe committed:
catalogue set up for now

Thekid2002 commented on src/app/webshop/catalogue/catalogue.component.html:
```diff
@@ -0,0 +1,48 @@
+        <mat-card>
+          <mat-card-content>
+            <mat-form-field>
+              <mat-label>Søg</mat-label>
```
This should not be in a grid. Make a new component calles search component and make it standard + reusable. Perhaps an input for a list of filters and an output for search.

Thekid2002 commented on src/app/webshop/catalogue/catalogue.component.html:
```diff
@@ -0,0 +1,48 @@
+      <mat-grid-tile>
+        <mat-card>
+          <mat-card-content>
+            <p>Filtrer:</p>
```
Same as above

Thekid2002 commented on src/app/webshop/catalogue/catalogue.component.html:
```diff
@@ -0,0 +1,48 @@
+
+
+              <mat-card-content>
+<!--                route link to id instead of webshop: {{wine.id}}-->
```
Vi prøver at undgå kommentare i koden

Thekid2002 replied:
Html koden

Thekid2002 commented on src/app/webshop/catalogue/catalogue.component.html:
```diff
@@ -0,0 +1,48 @@
+
+              <mat-card-content>
+<!--                route link to id instead of webshop: {{wine.id}}-->
+                <div routerLink="/webshop" style="cursor:pointer" >
```
Routerlink add queryparameter ?id={{id}}

Thekid2002 commented on src/app/webshop/catalogue/catalogue.component.ts:
```diff
@@ -0,0 +1,41 @@
+})
+
+export class CatalogueComponent {
+  wineType: WineType[] = [
```
Should be an enum connected to win

Thekid2002 commented on src/app/webshop/catalogue/catalogue.component.ts:
```diff
@@ -0,0 +1,41 @@
+    { name: 'Peanut Noar', price: 23, image: 'assets/PeanutNoar.jfif', id:5 }
+  ];
+}
+// export class CatalogueComponent {
```
This comment should be removed

vjohns22 committed:
WIP: added search componenet and somewhat responsive column amount

Thekid2002 committed:
SW-99 * Fixed dumme spørgsmål

vjohns22 committed:
SW-99 * Slight update to include price in search function

vjohns22 committed:
SW-99 * Added filters WIP

There are some questionable implementation quirks in here, however, the sorting / filtering should be correct, there is just an issue of unresponsiveness; the event does not update instantly yet.

also centered the wine imgs btw <3

Thekid2002 committed:
SW-99 * Er i gang med at rydde op i catalogue

Thekid2002 committed:
SW-99 * Done fixing catalogue

Thekid2002 committed:
SW-99 * Floating

Thekid2002 committed:
SW-99 * Fixed shit
